tests/generator/NullabilityContextTests.cs

```diff
@@ -0,0 +1,276 @@
+using System;
+using System.Collections.Generic;
+using bgen;
+using NUnit.Framework;
+
+#nullable enable
+
+namespace GeneratorTests {
+
+	[TestFixture]
+	[Parallelizable (ParallelScope.All)]
+	public class NullabilityContextTests {
+
+		class ObjectTest {
+			public int NotNullableValueTypeField = 0;
+			public string NotNullableRefTypeField = string.Empty;
+			public int? NullableValueTypeField = 0;
+			public string? NullableRefTypeField = string.Empty;
+
+			public int NotNullableValueTypeProperty { get; set; }
+			public string NotNullableRefProperty { get; set; } = string.Empty;
+			public int? NullableValueTypeProperty { get; set; }
+			public string? NullableRefProperty { get; set; }
+
+			public void NotNullableValueTypeParameter (int myParam) { }
+			public void NotNullableRefParameter (string myParam) { }
+
+			public void NullableValueTypeParameter (int? myParam) { }
+			public void NullableRefTypeParameter (string? myParam) { }
+
+			public int NotNullableValueTypeReturn () => 0;
```

Comment:
Generic types:

```cs
public T Generic<T> () where T: notnull {}
public T Generic<T> () where T: class? {}
public T Generic<T> () where T: NSObject? {}
public T Generic<T> () where T: INativeObject? {}
```

Reply:
As we talked online:

WRONG: public T GenericNullableRefTypeConstrain<T> () where T : ObjectTest?
GOOD:  public T? GenericNullableRefTypeConstrain<T> () where T : ObjectTest